draft-bormann-asdf-instance-information.md

```diff
@@ -483,11 +483,57 @@ sdfObject:
 {:sdf #code-sdf-constructors
 title="Example for SDF model with constructors"}
 
-The alternative approach is shown in TODO.
+The alternative approach is shown in {{code-sdf-constructor-action}}.
 Here, the constructor is modeled as an `sdfAction` that contains the same set of parameters in its `sdfInputData`.
 
-(Add example here and discuss more pros and cons of the two formats.)
+While this approach has advantages – we do not need to introduce new keywords to achieve a similar functionality and can simply use a plain JSON object as the construction message – a few things in this example are still unclear, especially when it comes to the mapping of constructor parameters to target affordances in the model and the designation of parameters as context information.
+Lastly, it is currently unclear what kind of schema information should be provided for the action's `sdfOutputData` since a detailed schema without using `sdfRef` would require the modeler to repeat whole structure of the model, creating a lot of verbosity.
 
+~~~ sdf
+info:
+  title: Example document for SDF with actions as constructors for instantiation
+  version: '2019-04-24'
+  copyright: Copyright 2019 Example Corp. All rights reserved.
+  license: https://example.com/license
+namespace:
+  cap: https://example.com/capability/cap
+defaultNamespace: cap
+sdfObject:
+  temperatureSensor:
+    sdfProperty:
+      temperature:
+        description: Temperature value measure by this Thing's temperature sensor.
+        type: number
+        unit:
+          "$comment": Should schema information be settable via a constructor at all? This question might indicate that we need different kinds of constructors
+          type: string
+          # target: "#/sdfObject/temperatureSensor/sdfProperty/temperature/unit"
+    sdfAction:
```

Comment:
- Move `target` under `construct`
  - Link to question: What model is valid here?
- Remove `temperature` and required
- Use as basis for further discussion

Reply:
One thing I noticed here is that the example is now rather a factory for models instead of documents related to instances (with the `ipAdress` being an exception as a piece of context information). I guess the question would be what kinds of output we produce from the constructor – would it be an extended model and a proofshot? Or rather a mapping file and a proofshot?